pkgs/stdenv/generic/setup.sh

```diff
@@ -313,16 +313,16 @@ appendToVar() {
     fi
 
     # check if variable already exist and if it does then do extra checks
-    if declare -p "$1" 2> /dev/null | grep -q '^'; then
-        type="$(declare -p "$1")"
-        if [[ "$type" =~ "declare -A" ]]; then
-            echo "appendToVar(): ERROR: trying to use appendToVar on an associative array, use variable+=([\"X\"]=\"Y\") instead." >&2
-            return 1
-        elif [[ "$type" =~ "declare -a" ]]; then
-            useArray=true
-        else
-            useArray=false
-        fi
+    if type=$(declare -p "$1" 2> /dev/null); then
+        case "${type#* }" in
```

Comment:
Since Bash 4.4 we can use `${var@a}` to check variable attributes (e.g. `[[ ${var@a} == *a* ]]` to check that it’s an array, or equivalent pattern in `case` statement), and use the good ol’ `[[ -z ${var+x} ]]` to check that variable is defined. See https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Also applies to other cases where we check variable attributes.

Note that currently stdenv requires at least Bash ≥4 but I don’t think we require a particular minor version. Perhaps it’s time to explicitly require at least 4.4 (released in 2016) (use `"${BASH_VERSINFO[1]}"` to check that since `BASH_VERSINFO` is actually an array, see https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-BASH_005fVERSINFO). See also https://mywiki.wooledge.org/BashFAQ/061